internal/gitinterface/blob_test.go

```diff
@@ -0,0 +1,88 @@
+package gitinterface
+
+import (
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+
+	"github.com/adityasaky/gittuf/internal/common"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReadBlob(t *testing.T) {
+	testDir, err := common.CreateTestRepository()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	readContents := []byte("test file read")
+	if err := os.WriteFile(path.Join(testDir, "test-file-read"), readContents, 0644); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := os.Chdir(testDir); err != nil {
+		t.Fatal(err)
+	}
+
+	cmd := exec.Command("git", "hash-object", "-w", "test-file-read")
+	if err := cmd.Run(); err != nil {
+		t.Fatal(err)
+	}
+
+	repo, err := common.GetRepositoryHandler()
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
Instead of doing all of this, you could do the in-memory setup + WriteBlob, then verify you get the same content on read.